src/main/java/org/opensearch/security/support/ConfigConstants.java

```diff
@@ -382,6 +385,10 @@ public enum RolesMappingResolution {
     // Variable for initial admin password support
     public static final String OPENSEARCH_INITIAL_ADMIN_PASSWORD = "OPENSEARCH_INITIAL_ADMIN_PASSWORD";
 
+    // Resource sharing feature-flag
+    public static final String OPENSEARCH_RESOURCE_SHARING_ENABLED = SECURITY_SETTINGS_PREFIX + "resource_sharing.enabled";
+    public static final boolean OPENSEARCH_RESOURCE_SHARING_ENABLED_DEFAULT = true;
```

Comment:
I see this is by design but just want to discuss with other reviewers - is there any concern with having this feature be default enabled? I do not see too much concern with it since it would be opt in only via implementing the SPI.

Reply:
experimental feature flags should be false while experimental. We can certainly change default values in the future in major releases.

Reply:
This feature flag is for initial versions. It is intended to be permanently enabled.

On another hand, we can simply say that if security is enabled then we assume that the feature is enabled. For now, i have kept these as separate.

Reply:
I am also a bit worried regarding the "experimental" status. I tried to find some guidelines regarding experimental features.

There's this page in the docs: https://opensearch.org/docs/latest/install-and-configure/configuring-opensearch/experimental/ ... however it is not too helpful.

There is another issue where a discussion takes place whether experimental features should be enabled or disabled: https://github.com/opensearch-project/OpenSearch/issues/13411 .. the consensus there is that experimental features should be not enabled by default.

So, I guess we have two choices here:

- Disable the feature by default
- Do not mark the feature as experimental

Reply:
For this iteration, I will update it to be disabled by default.

Reply:
just  a note. If this feature is enabled it will be enabled for all plugins.